src/jdk.jshell/share/classes/jdk/jshell/execution/impl/ConsoleImpl.java

```diff
@@ -395,7 +395,9 @@ public synchronized void write(int b) throws IOException {
                 buffer = Arrays.copyOf(buffer, 2 * buffer.length);
             }
 
-            buffer[bp++] = b;
+            // Can be negative because directly casted from byte.
+            // java.io.OutputStream.write(int b) stipulates "The 24 high-order bits of b are ignored."
+            buffer[bp++] = b & 0xff;
```

Comment:
I think the cause is that `OutputStream.write(byte[], int, int)` provides negative bytes. I recommend you to update the "can be negative" comment above to be like:

```
// Can be negative because widening from byte in write(byte[], int, int).
```

Also note, there is another usage of `b` below in `case READ_CHARS ->` but I think it is also a bug; it should be `readInt(1)`.

Reply:
> I think the cause is that OutputStream.write(byte[], int, int) provides negative bytes.

I seem to recall that perhaps so. Your suggestion is better.

> it should be readInt(1).

Do you mean `int len = readInt(b);`? I will update it but unfortunately I have no idea which is correct. Do you have any idea to test that the fix is correct?
I could not tell the difference in jshell by playing with some input.

Reply:
Yep, that readInt is probably fine because the buffer is large enough, plus the result is unused, so there was no observable consequence.

Also in principle, we will not approve patches unless oca is cleared; meanwhile please update the 2024 last updated year in the license header of 2 files to 2025. (I thought this rule was mentioned in the guide, but apparently it wasnt!)

Reply:
Which notation is the best?

- 2023-2025 (uses the ASCII hypen)
- 2023–2025 (uses U+2013 EN dash)
- 2023, 2024, 2025

Reply:
Yep, `Copyright (c) 2023, 2025, Oracle` is correct.

Reply:
Thank you. Does the current change look good to you?

Reply:
Yes. But I will get a more professional jshell reviewer to double check once the oca is resolved.